public_html/wp-content/themes/pattern-directory/src/store/test/resolvers.js

```diff
@@ -5,9 +5,10 @@ import apiPatterns from './fixtures/patterns';
 import apiCategories from './fixtures/categories';
 import apiPatternFlagReasons from './fixtures/pattern-flag-reasons';
 import { getCategories, getFavorites, getPatternFlagReasons, getPatternsByQuery } from '../resolvers';
+import { PER_PAGE } from '../utils';
 
 describe( 'getPatternsByQuery', () => {
-	it( 'yields with the requested patterns', async () => {
+	it( 'yields with the requested patterns & query meta', async () => {
```

Comment:
Since the query is most important should we write a test to expose the error state for future error handling?

Reply:
I'm not sure what you mean here, can you explain?

Reply:
Since we are catching/disregarding errors when we query the API, smart error handling will be tricky since it's not clear what the state will look like when a request fails. I'm wondering if we should write a test, intercept the API, fail it however we want, and expose what the state will look like in that scenario. It would give future developers some information on how to improve query retries or help provide better user messaging if we don't get to that before MVP launch.

For example, it appears to me, by looking at the code that if the query fails we should get:

```
{
 page: 1,
 patterns: undefined,
 total: undefined,
 totalPages: undefined,
}
```
A test would communicate that clearly.

Reply:
I'm not stuck on this though, feel free to merge without it.

Reply:
Oh I always planned on us getting to this before launch, just not necessarily here 😁  In fact, i ended up doing some basics in #113.